### PR TITLE
GG-24257: Add distributed actions by compute execution.

### PR DESCRIPTION
**Pull request by smoldachev**

### CODE REVIEW

File: modules/gmc-agent/src/main/java/org/apache/ignite/agent/service/action/ExecuteActionTask.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    /** {@inheritDoc} */
+    @Override public Map<? extends ComputeJob, ClusterNode> map(List<ClusterNode> subgrid, Request arg) throws IgniteException {
+        return subgrid.stream().collect(Collectors.toMap(n -> new ExecuteActionJob(arg), identity()));
+    }
+
+    /** {@inheritDoc} */
+    @Override public Response reduce(List<ComputeJobResult> results) throws IgniteException {
+        ComputeJobResult computeJobRes = results.get(0);
```
Comment (nva) [inline modules/gmc-agent/src/main/java/org/apache/ignite/agent/service/action/ExecuteActionTask.java]:
```suggestion
        ComputeJobResult computeJobRes = F.first(results);
```

File: modules/gmc-agent/src/main/java/org/apache/ignite/agent/service/action/ExecuteActionTask.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        /**
+         * @param req Request.
+         */
+        public ExecuteActionJob(Request req) {
+            this.req = req;
+        }
+
+        /** {@inheritDoc} */
+        @Override public void cancel() {
+            isCanceled = true;
```
Comment (nva) [inline modules/gmc-agent/src/main/java/org/apache/ignite/agent/service/action/ExecuteActionTask.java]: Can't find any usage

File: modules/gmc-agent/src/main/java/org/apache/ignite/agent/service/action/DistributedActionService.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+import static org.apache.ignite.agent.StompDestinationsUtils.buildActionResponseDest;
+import static org.apache.ignite.agent.dto.action.ActionStatus.FAILED;
+import static org.apache.ignite.agent.dto.action.ActionStatus.RUNNING;
+import static org.apache.ignite.agent.utils.AgentUtils.getErrorCode;
+
+/**
+ * Distributed action service.
+ */
+public class DistributedActionService {
```
Comment (nva) [inline modules/gmc-agent/src/main/java/org/apache/ignite/agent/service/action/DistributedActionService.java]: Incorrect name for this service, action will be executed only on one node

File: modules/management-console-agent/src/main/java/org/apache/ignite/agent/service/action/DistributedActionService.java [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+        mgr.send(buildActionJobResponseDest(clusterId, res.getRequestId()), res);
+    }
+
+    /**
+     * @param reqId Request id.
+     * @param e Exception.
+     * @return Task response.
+     */
+    private TaskResponse sendFailedTaskResponse(UUID reqId, Throwable e) {
```
Comment (nva) [inline modules/management-console-agent/src/main/java/org/apache/ignite/agent/service/action/DistributedActionService.java]: Why we send job failed if task doesn't execute?

File: modules/management-console-agent/src/main/java/org/apache/ignite/agent/service/action/ExecuteActionTask.java [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+        if (res.getException() == null)
+            srvc.sendJobResponse(jobRes);
+        else {
+            log.error(
+                String.format("Failed to execute job, send response with error: [reqId=%s]", reqId),
+                res.getException()
+            );
+
+            srvc.sendJobResponse(convertToErrorJobResponse(reqId, consistentId, res.getException()));
```
Comment (nva) [inline modules/management-console-agent/src/main/java/org/apache/ignite/agent/service/action/ExecuteActionTask.java]: Can we ignore other job results?

File: modules/management-console-agent/src/main/java/org/apache/ignite/agent/dto/action/RequestDeserializer.java [hunk 6]
```diff
@@ -64,7 +67,7 @@ public RequestDeserializer() {
             }
         }
         catch (Exception e) {
-            req = new InvalidRequest().setCause(e).setId(id).setAction(act);
+            req = new InvalidRequest().setCause(new IllegalArgumentException(e)).setId(id).setAction(act);
```
Comment (akuznetsov-gridgain) [inline modules/management-console-agent/src/main/java/org/apache/ignite/agent/dto/action/RequestDeserializer.java]: Why do we need to wrap exception with IllegalArgumentException?
And what if `e` already will be `IllegalArgumentException` (see line 57)?

File: modules/management-console-agent/src/main/java/org/apache/ignite/agent/dto/action/TaskResponse.java [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+    /**
+     * @param status Status.
+     * @return This for chaining method calls.
+     */
+    public TaskResponse setStatus(Status status) {
+        this.status = status;
+        return this;
+    }
+}
```
Comment (akuznetsov-gridgain) [inline modules/management-console-agent/src/main/java/org/apache/ignite/agent/dto/action/TaskResponse.java]: Missing `toString()`

File: modules/management-console-agent/src/main/java/org/apache/ignite/agent/dto/action/query/QueryArgument.java [hunk 8]
```diff
@@ -42,7 +41,7 @@
     private boolean enforceJoinOrder;
 
     /** Target node ID. */
-    private UUID targetNodeId;
+    private boolean isLoc;
```
Comment (akuznetsov-gridgain) [inline modules/management-console-agent/src/main/java/org/apache/ignite/agent/dto/action/query/QueryArgument.java]: Javadocs is not related to this field.

**Pull request closed without merging**